src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -651,13 +664,26 @@ - (void)invalidate
 
 - (void)nodeMayBeAdvertisingOperational
 {
+    assertChipStackLockedByCurrentThread();
+
     MTR_LOG_DEFAULT("%@ saw new operational advertisement", self);
 
+    [self _triggerResubscribeWithReason:"operational advertisement seen"
+                    nodeLikelyReachable:YES];
+}
+
+// Trigger a resubscribe as needed.  nodeLikelyReachable should be YES if we
+// have reason to suspect the node is now reachable, NO if we have no idea
+// whether it might be.
+- (void)_triggerResubscribeWithReason:(const char *)reason nodeLikelyReachable:(BOOL)nodeLikelyReachable
```

Comment:
If `nodeLikelyReachable` should this also reset `_lastSubscriptionAttemptWait` to 0?

Reply:
@jtung-apple  That might be reasonable, yes, but I'd like to do that in a separate PR, since it's a change to existing behavior....